validator/client/beacon-api/submit_aggregate_selection_proof.go

```diff
@@ -47,40 +114,56 @@ func (c *beaconApiValidatorClient) submitAggregateSelectionProof(
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to calculate attestation data root")
 	}
+	return attestationDataRoot[:], nil
+}
 
-	aggregateAttestationResponse, err := c.aggregateAttestation(ctx, in.Slot, attestationDataRoot[:])
-	if err != nil {
-		return nil, err
-	}
-
-	var attData *structs.Attestation
-	if err := json.Unmarshal(aggregateAttestationResponse.Data, &attData); err != nil {
-		return nil, errors.Wrap(err, "failed to unmarshal aggregate attestation data")
-	}
+func (c *beaconApiValidatorClient) aggregateAttestation(
+	ctx context.Context,
+	slot primitives.Slot,
+	attestationDataRoot []byte,
+	committeeIndex primitives.CommitteeIndex,
+) (*structs.AggregateAttestationResponse, error) {
+	params := url.Values{}
+	params.Add("slot", strconv.FormatUint(uint64(slot), 10))
+	params.Add("attestation_data_root", hexutil.Encode(attestationDataRoot))
+	params.Add("committee_index", strconv.FormatUint(uint64(committeeIndex), 10))
+	endpoint := buildURL("/eth/v2/validator/aggregate_attestation", params)
 
-	aggregatedAttestation, err := convertAttestationToProto(attData)
+	var aggregateAttestationResponse structs.AggregateAttestationResponse
+	err := c.jsonRestHandler.Get(ctx, endpoint, &aggregateAttestationResponse)
```

Comment:
I think i need to add a header to this too for eth consensus version https://ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/Validator/getAggregatedAttestationV2